python/sglang/srt/models/deepseek_v2.py

```diff
@@ -1842,12 +1879,27 @@ def forward(
             hidden_states, residual, forward_batch
         )
 
-        hidden_states = self.mlp(hidden_states, forward_batch)
-
-        hidden_states, residual = self.layer_communicator.postprocess_layer(
-            hidden_states, residual, forward_batch
+        can_fuse_mlp_allreduce = (
```

Comment:
Could you try to cache the results and reuse them across all layers?
Basically, do not put these checks in the critical path

Reply:
Make sense, I will try to cache it.